site/server/app/com/fortysevendeg/exercises/services/interpreters/interpreters.scala

```diff
@@ -48,12 +102,24 @@ trait InterpreterInstances {
     override def pureEval[A](x: Eval[A]): Task[A] =
       Task.fork(Task.delay(x.value))
 
-  }
+    override def raiseError[A](e: Throwable): Task[A] =
+      Task.fail(e)
 
+    override def handleErrorWith[A](fa: Task[A])(f: Throwable ⇒ Task[A]): Task[A] =
+      fa.handleWith({ case x ⇒ f(x) })
+  }
 }
 
 /** Production based interpreters lifting ops to the effect capturing scalaz.concurrent.Task **/
-object ProdInterpreters extends Interpreters[Task]
+object ProdInterpreters extends Interpreters[Task] {
+  implicit class FreeOps[A](f: Free[ExercisesApp, A]) {
+
+    /** Run this Free structure folding it's ops to an effect capturing task */
+    def runTask(implicit T: Transactor[Task]): Throwable \/ A = {
```

Comment:
Let's return an `Xor` here so we don't leak the Scalaz types since we are using cats elsewhere